libs/sdk-backend-bear/src/convertors/dateFormatting/dateFormatter.ts

```diff
@@ -1,19 +1,53 @@
 // (C) 2020 GoodData Corporation
-import { IResultHeader, isResultAttributeHeader } from "@gooddata/sdk-backend-spi";
+import {
+    IAttributeDescriptor,
+    IDimensionDescriptor,
+    IResultHeader,
+    isAttributeDescriptor,
+    isResultAttributeHeader,
+} from "@gooddata/sdk-backend-spi";
 import { createDateValueFormatter } from "./dateValueFormatter";
-import { DateFormat } from "./dateValueParser";
+import { DateFormat, DEFAULT_DATE_FORMAT } from "./dateValueParser";
 import { createDefaultDateFormatter } from "./defaultDateFormatter";
 
-export function transformDateFormat(resultHeader: IResultHeader, dateFormat?: DateFormat): IResultHeader {
-    if (!isResultAttributeHeader(resultHeader) || !dateFormat) {
+export function findDateAttributeUri(dimensions: IDimensionDescriptor[]): string | undefined {
+    const dateAttrDescriptor = dimensions
+        .reduce((attrDimensions: IAttributeDescriptor[], dimensionItem: IDimensionDescriptor) => {
+            attrDimensions.push(...dimensionItem.headers.filter(isAttributeDescriptor));
+            return attrDimensions;
+        }, [])
+        .find((attrDescriptor: IAttributeDescriptor): boolean =>
+            attrDescriptor.attributeHeader.identifier.endsWith("mmddyyyy"),
```

Comment:
Is it safe to assume that this convention is always correct?

Reply:
I've searched through the urn files in gdc-datedim/metadata/urn:*** (https://github.com/gooddata/gdc-datedim/search?q=type%3A+GDC.time.day_us) and found that:

- For all files other than "urn:custom_v2:date.yaml" and "urn:custom:date.yaml", all date display forms of type "GDC.time.day_us" have identifiers ending with "mmddyyyy".

- For two files "urn:custom_v2:date.yaml" and "urn:custom:date.yaml", the date display forms of type "GDC.time.day_us" have identifiers ending with "mm_dd_yyyy".

As far as I know, there is no use case to change the identifier of a date display form, meaning that we can know whether an attribute descriptor is a date by checking if its identifier ends with "mmddyyyy" or "mm_dd_yyyy". (Checking for the missing case "mm_dd_yyyy" has been added, please help me review again).

Reply:
Can you please confrim this with someone from backend? Just to be sure.

Reply:
Thanks for you suggestion. I've confirmed with @david-kubecka and he said that "the only correct way how to identify date attributes is by the attribute "type", e.g. GDC.time.day_us".

I've created a PR in gdc-webapp (https://github.com/gooddata/gdc-webapp/pull/4183) and also updated this PR so that date attributes will be identified if its type is "GDC.time.day_us". Please help me review again, thanks.

Reply:
Awesome, thank you very much for handling this 👍